platform/android/export/export.cpp

```diff
@@ -2536,63 +2549,111 @@ class EditorExportPlatformAndroid : public EditorExportPlatform {
-					EditorNode::get_singleton()->show_warning(vformat(TTR("Android build version mismatch:\n   Template installed: %s\n   Godot Version: %s\nPlease reinstall Android build template from 'Project' menu."), version, VERSION_FULL_CONFIG));
-					return ERR_UNCONFIGURED;
-				}
+			//test that installed build version is alright
+			FileAccessRef f = FileAccess::open("res://android/.build_version", FileAccess::READ);
+			if (!f) {
+				EditorNode::get_singleton()->show_warning(TTR("Trying to build from a custom built template, but no version info for it exists. Please reinstall from the 'Project' menu."));
+				return ERR_UNCONFIGURED;
 			}
+			String version = f->get_line().strip_edges();
+			if (version != VERSION_FULL_CONFIG) {
+				EditorNode::get_singleton()->show_warning(vformat(TTR("Android build version mismatch:\n   Template installed: %s\n   Godot Version: %s\nPlease reinstall Android build template from 'Project' menu."), version, VERSION_FULL_CONFIG));
+				return ERR_UNCONFIGURED;
+			}
+			String sdk_path = EDITOR_GET("export/android/custom_build_sdk_path");
+			ERR_FAIL_COND_V_MSG(sdk_path == "", ERR_UNCONFIGURED, "Android SDK path must be configured in Editor Settings at 'export/android/custom_build_sdk_path'.");
 
 			// TODO: should we use "package/name" or "application/config/name"?
 			String project_name = get_project_name(p_preset->get("package/name"));
-			// instead of calling _fix_resources
-			Error err = _create_project_name_strings_files(p_preset, project_name);
+			err = _create_project_name_strings_files(p_preset, project_name); //project name localization.
 			if (err != OK) {
 				EditorNode::add_io_error("Unable to overwrite res://android/build/res/*.xml files with project name");
 			}
 			// Copies the project icon files into the appropriate Gradle project directory.
 			_copy_icons_to_gradle_project(p_preset, main_image, foreground, background);
 			// Write an AndroidManifest.xml file into the Gradle project directory.
 			_write_tmp_manifest(p_preset, p_give_internet, p_debug);
-			//build project if custom build is enabled
-			String sdk_path = EDITOR_GET("export/android/custom_build_sdk_path");
-
-			ERR_FAIL_COND_V_MSG(sdk_path == "", ERR_UNCONFIGURED, "Android SDK path must be configured in Editor Settings at 'export/android/custom_build_sdk_path'.");
-
 			_update_custom_build_project();
+			//stores all the project files inside the Gradle project directory. Also includes all ABIs
+			if (!apk_expansion) {
+				DirAccess *da_res = DirAccess::create(DirAccess::ACCESS_RESOURCES);
+				if (da_res->dir_exists("res://android/build/assets")) {
+					DirAccess *da_assets = DirAccess::open("res://android/build/assets");
+					da_assets->erase_contents_recursive();
+					da_res->remove("res://android/build/assets");
+				}
+				err = export_project_files(p_preset, rename_and_store_file_in_gradle_project, NULL, ignore_so_file);
+			} else {
+				Error err = save_apk_expansion_file(p_preset, p_path);
```

Comment:
You're redeclaring `Error err` hence shadowing the variable declared outside this block.